python/lsst/ts/gis/csc.py

```diff
@@ -103,27 +103,27 @@ async def handle_summary_state(self):
         """
         if self.disabled_or_enabled:
             if self.simulation_mode and self.mock_server is None:
-                child = ModbusSlaveContext(hr=ModbusSequentialDataBlock(0, [0] * 30))
-                context = ModbusServerContext(slaves=child, single=True)
-                self.mock_server = ModbusTcpServer(
-                    context=context, address=("127.0.0.1", 15020)
-                )
-                self.log.debug("Starting mock server.")
-                self.mock_server_task = asyncio.create_task(
-                    self.mock_server.serve_forever()
-                )
-                await self.mock_server.serving
-                self.log.debug("Mock server started.")
+                self.mock_server = ModbusSimulatorServer(json_file=FILE)
+                self.log.info("Starting mock server.")
+                if self.mock_server_task.done():
+                    self.mock_server_task = asyncio.create_task(
+                        self.mock_server.run_forever()
+                    )
+                # In pymodbus 3.5, serve_forever added an await serving which
```

Comment:
You have pinned the pymodbus version to `>= 3.5` so please remove this `if` completely.